### ISSUE TITLE
*-firefox extension packages need updating to new method

### ISSUE THREAD

**Issue by bcurran3** [bcurran3/ChocolateyPackages_CCR#63]
Newer versions of Firefox have changed how local extensions get installed and rendered all of my "-firefox" extensions obsolete.

I spent some time trying to figure out a new way via some info I found on the web, but I wasn't successful.

Any links on how to do it would be appreciated.

**Comment by teknowledgist** [comment]
Are you asking about packages for actual Firefox extensions, or do you have a Chocolatey extension for helping with Firefox extensions? 

If the former:  did you see [Mozilla's documentation]( https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Distribution_options/Add-ons_in_the_enterprise)?  It looks like the registry method should be do-able for signed extensions.  The extension wouldn't be available until Firefox restarts, and the user would need to "OK" or enable it, but it could be system-wide, and install/uninstall-able while Firefox is running.

**Comment by bcurran3** [comment]
Former. I've got a bunch of Chocolatey packages that install Firefox extensions but they stopped working when Firefox changed how things worked. I investigated but couldn't find the right location to unzip them and make them work (possibly didn't restart Firefox). 

The packages are all easily editable templates. Example: 
https://chocolatey.org/packages/google-shortcuts-firefox

I guess I'll go back and investigate again.

**Comment by koolhand** [comment]
Making some notes to see if I can help...

### Why

* [Mozilla Add-Ons Blog - Upcoming changes to extension sideloading](https://blog.mozilla.org/addons/2019/10/31/firefox-to-discontinue-sideloaded-extensions/)

### How

* [Mozilla KB - Deploying Firefox With Extensions](https://support.mozilla.org/en-US/kb/deploying-firefox-with-extensions)
* [ExtensionWorkshop.com - Sideloading add-ons](https://extensionworkshop.com/documentation/publish/distribute-sideloading/)

### What

* [colorfultabs-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/colorfultabs-firefox)
* [flagfox-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/flagfox-firefox)
* [gmailnotifier-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/gmailnotifier-firefox)
* [google-shortcuts-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/google-shortcuts-firefox)
* [https-everywhere-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/https-everywhere-firefox)
* [ietab2-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/ietab2-firefox)
* [trafficlight-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/trafficlight-firefox)
* [wot-firefox](https://github.com/bcurran3/ChocolateyPackages/tree/master/wot-firefox)

**Comment by koolhand** [comment]
Hi @bcurran, I did some reading this morning,

The "sideloading" method of installing at the machine-level just isn't supported anymore. Mozilla plugins are kinda meant to install into user profile, not at the machine / program-files level. And they self-update using Firefox's own mechanism, in user-land.


You could try switching to use Firefox's [Group Policy reg keys](https://github.com/mozilla/policy-templates#extensions) which forcibly install XPIs, which are intended for a corporate environment where lots of machines are kept in the same config. That would force it to install into every logged on user's profile, separately. But... really you'd be working "against the grain" of what Firefox natively supports.

So bottom line: I think you should remove the *-firefox extension packages from your repo.

Pull request #203 will do that if you're interested.

### Removed:

* colorfultabs-firefox
* flagfox-firefox
* gmailnotifier-firefox
* google-shortcuts-firefox
* https-everywhere-firefox
* ietab2-firefox
* trafficlight-firefox
* wot-firefox

**Comment by bcurran3** [comment]
Yeah, I might still go that route for install. Just don't have the time or urgency to do any time soon.
No need to remove from repo, if if these lay dormant or I later kill.

**Comment by bcurran3** [comment]
Closing this as I am no longer creating or maintaining Chocolatey packages. Sorry, out
of my control.

Hopefully you've enjoyed my work and found it useful.

[![Bella Ciao!](https://img.youtube.com/vi/9ao4FEaDGhQ/0.jpg)](https://www.youtube.com/watch?v=9ao4FEaDGhQ)

**Comment by koolhand** [comment]
Thanks. You provided a public service for free, for years, and it was very much appreciated, I’m sure it was a huge amount of work. I hope letting these packages go is a weight off your mind! Cheers to you 🍻

**Comment by bcurran3** [comment]
Much appreciated kind words Luke, thanks.
